### PR TITLE
NO-ISSUE: Update doc: nmstate and agent conditions

### PR DESCRIPTION
**Pull request by rollandf**

### CODE REVIEW

**Comment by rollandf** [comment]
/hold
Until #1457 is merged.

File: docs/Kube-API.md
Comment (nmagnezi) [inline docs/Kube-API.md:29]:
```suggestion
- InfraEnv CR: add a label to nmStateConfigLabelSelector with a user defined name and value.
```

File: docs/Kube-API.md
Comment (nmagnezi) [inline docs/Kube-API.md:32]:
```suggestion
Upon InfraEnv creation, the InfraEnv controller will search by label+value for matching NMState resources and construct a config to be sent as StaticNetworkConfig as a part of ImageCreateParams. The backend does all validations, and currently, there is no handling of configuration conflicts (e.g., two nmstate resources using the same MAC address).
```

Comment (nmagnezi) [reply docs/Kube-API.md:32]: ditto

File: docs/Kube-API.md
Comment (nmagnezi) [inline docs/Kube-API.md:34]:
```suggestion
The InfraEnv controller will watch for NMState config creation/changes and search for corresponding InstallEnv resources to reconcile since we need to regenerate the image for those.
```

Comment (nmagnezi) [reply docs/Kube-API.md:34]: ditto

File: docs/Kube-API.md
Comment (nmagnezi) [inline docs/Kube-API.md:63]:
```suggestion
$ kubectl get agents.agent-install.openshift.io -n assisted-installer  -o=jsonpath='{range .items[*]}{"\n"}{.status.inventory.hostname}{"\n"}{range .status.conditions[*]}{.type}{"\t"}{.message}{"\n"}{end}'
```

Comment (nmagnezi) [reply docs/Kube-API.md:63]: super-duper nit, I know.
this is very cool btw :)

File: docs/Kube-API.md
Comment (nmagnezi) [inline docs/Kube-API.md:48]: for consistency, can you please add `sh` syntax highlighting as you did in line 63?

**Review by nmagnezi — COMMENTED** [review]
looks good to me.
will let other people a chance to review as well.

**Review by nmagnezi — APPROVED** [review]

**Pull request merged**
